private/buf/cmd/buf/command/registry/registrylogin/registrylogin.go

```diff
@@ -137,6 +138,7 @@ func run(
 	if err != nil {
 		return err
 	}
+	token = stringutil.TrimLines(token)
```

Comment:
I know you mentioned this in the summary, but we could return an error here and include an error message with something like `"token contains a newline - did you mean to use 'echo -n'?"`

Trimming the newlines is probably harmless, I'm just always weary of silently adapting the user's input if we can avoid it. Let me know what you think - I'm not strongly opposed.

Reply:
Ah, good to know about `stringutil.TrimLines`. I definitely think this is one of those few Postel's law situations where liberally accepting is not likely to cause problems and makes for a much better user experience. Perhaps a comment to that effect is warranted? Otherwise this LGTM.

Reply:
I suppose we also have the same problem with username, potentially?

Reply:
> our stringutil.TrimLines(..) also calls strings.TrimSpace(..). Maybe strings.TrimSpace(strings.TrimSuffix(s, "\n")) would be sufficient, but iirc we tend to prefer our utils.

FWIW `strings.TrimSpace` also strips newlines, as they are considered white space per `unicode.IsSpace`.

Reply:
Ah yes, forgot about that. Switched to `strings.TrimSpace`, great suggestion! Also added a comment and an extra check for empty tokens.